### ISSUE TITLE
[telemetry] Internal error counters in InternalErrorHandler

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#46]
## Goal

Count internal errors in `ph-oton-core` and record them on the active ph-telemetry span.

An internal error is by definition the thing an operator most wants alerting on, and today it produces a log line and (maybe) an e-mail — nothing a monitoring system can threshold.

## Where

`ph-oton-core/src/main/java/com/helger/photon/core/interror/InternalErrorHandler.java`

```java
static String handleInternalError (final boolean bSendEmail, ...)
```

This is the central funnel — it logs `"handleInternalError " + sErrorID` with the throwable, fills in `InternalErrorMetadata`, optionally sends the notification mail and optionally persists an XML file.

Two suppression paths in the same class are worth surfacing:
- the occurrence-count limit: `"Not sending internal error mail, because this error occurred N times"`
- missing configuration: `"Not sending internal error mail, because 'sender'/'receiver'/'SMTP settings' is not set!"`

An operator who has silently mis-configured the sender address currently gets **no** notification and **no** signal that notifications are broken. A counter fixes that.

## What to emit

Extend `com.helger.photon.core.CCoreTelemetry` (create it here if the UI-rendering issue has not landed yet):

| Constant | Value |
|---|---|
| `METRIC_INTERNAL_ERRORS` | `photon.internalerror.count` |
| `METRIC_INTERNAL_ERROR_MAILS_SUPPRESSED` | `photon.internalerror.mail.suppressed` |
| `ATTR_INTERNAL_ERROR_TYPE` | `photon.internalerror.type` (throwable **class name** — see cardinality note) |
| `ATTR_INTERNAL_ERROR_ID` | `photon.internalerror.id` (**span only**) |
| `ATTR_INTERNAL_ERROR_SUPPRESS_REASON` | `photon.internalerror.suppress.reason` |
| `UNIT_ERROR` | `{error}` |

- **Counter** `photon.internalerror.count`, attribute `type` = the throwable's class name, or the string `none` when no throwable is present.
- **Counter** `photon.internalerror.mail.suppressed`, attribute `reason` — a small fixed enum of string constants: `occurrence-limit`, `no-sender`, `no-receiver`, `no-smtp-settings`. Define them as constants in `CCoreTelemetry`; do not build the string ad hoc at each call site.
- **Span**: no new span is started. Instead, record the throwable on the *currently active* span so the error shows up in whatever trace it happened in. `Telemetry` has no "current span" accessor, so this means calling `recordException` on a span the caller passes in, or simply skipping the span part. **Decide during implementation** and keep it simple — the counters are the deliverable; the span integration is optional and must not drive an API addition to ph-telemetry.

## Cardinality

- Throwable **class name** is bounded in practice (a few dozen distinct types) and is the right dimension.
- The throwable **message** is unbounded (it usually embeds IDs, paths, SQL) and must **never** be a metric attribute.
- The internal error ID is unique per occurrence — span attribute only, never a metric.

## POM

`ph-oton-core` gets `ph-telemetry` transitively via `ph-oton-mgrs`, but declare it directly since it is used directly (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Every call to `handleInternalError (...)` produces exactly one `photon.internalerror.count` increment.
- [ ] All four mail-suppression paths increment `photon.internalerror.mail.suppressed` with a constant reason string.
- [ ] No throwable message and no error ID on any metric attribute.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.5.1**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-core/.../core/CCoreTelemetry.java` | extended - 2 metric names, 2 attribute keys, 5 attribute value constants, `UNIT_ERROR` |
| `ph-oton-core/.../core/interror/InternalErrorMetrics.java` | new, public - 2 counters |
| `ph-oton-core/.../core/interror/InternalErrorTelemetry.java` | new, package-private emitter |
| `ph-oton-core/.../core/interror/InternalErrorHandler.java` | modified - 5 call sites |

### Emission

* Counter `photon.internalerror.count`, incremented once at the top of `handleInternalError (...)` right after the `LOGGER.error (...)` line, so **every** call is counted regardless of debug mode, mail settings or custom handlers. Attribute `photon.internalerror.type` = `t.getClass ().getName ()`, or the constant `none` when no throwable is present.
* Counter `photon.internalerror.mail.suppressed`, attribute `photon.internalerror.suppress.reason`, at all four suppression sites in `_sendInternalErrorMailToVendor (...)`:

| Log line | Reason constant |
|---|---|
| `...because this error occurred N times` | `CCoreTelemetry.VALUE_SUPPRESS_OCCURRENCE_LIMIT` = `occurrence-limit` |
| `...because 'sender' is not set!` | `VALUE_SUPPRESS_NO_SENDER` = `no-sender` |
| `...because 'receiver' is not set!` | `VALUE_SUPPRESS_NO_RECEIVER` = `no-receiver` |
| `...because 'SMTP settings' is not set!` | `VALUE_SUPPRESS_NO_SMTP_SETTINGS` = `no-smtp-settings` |

The reason strings are constants in `CCoreTelemetry`, not built ad hoc at the call sites, as required. This is the part that closes the real gap: an operator who mis-configured the sender address previously got no notification *and* no signal that notifications are broken.

### Cardinality

* Throwable **class name** is the metric dimension - bounded in practice.
* The throwable **message** is never used anywhere; it typically embeds IDs, paths and SQL.
* The unique internal error ID is never used either - see below.

### The span part: decided against

The issue left this open and said the counters are the deliverable. No span integration was added:

`Telemetry` deliberately exposes no "current span" accessor, so recording the throwable on whatever trace the error happened in would either require threading a span through `handleInternalError (...)`'s already twelve-parameter signature, or an API addition to ph-telemetry - which the issue explicitly ruled out. The exception is in any case already recorded on the surrounding `photon.page.request` / `photon.api.invoke` / `photon.ajax.invoke` span by the Tier 1 instrumentation, which is where an operator would look for it.

Consequently `ATTR_INTERNAL_ERROR_ID` was **not** added - a span-only attribute constant with no use site would be dead code. Same reasoning as for `ATTR_LOGIN_USER_ID` in #43.

### Verification

Full reactor `mvn install` over all 23 modules: BUILD SUCCESS, no test failures. No unit test was added - the issue's acceptance criteria did not ask for one, and `handleInternalError (...)` reaches SMTP and file persistence. `README.md` news bullet added under `v10.5.1 - work in progress`.
